src/linker/Linker.Steps/SweepStep.cs

```diff
@@ -72,6 +80,71 @@ void RemoveUnusedAssembly (AssemblyDefinition assembly)
 			}
 		}
 
+		void UpdateAssembliesReferencingRemovedAssembly (AssemblyDefinition assembly)
+		{
+			foreach (var a in assemblies) {
+				var action = Annotations.GetAction (a);
+				switch (action) {
+				case AssemblyAction.Skip:
+				case AssemblyAction.Delete:
+				case AssemblyAction.Link:
+				case AssemblyAction.Save:
+					continue;
+
+				case AssemblyAction.Copy:
+				case AssemblyAction.CopyUsed:
+				case AssemblyAction.AddBypassNGen:
+				case AssemblyAction.AddBypassNGenUsed:
+					// Assembly was removed in the output but it's referenced by
+					// other assembly with action which does not update references
+					if (!HasReferenceTo (a, assembly))
+						continue;
+
+					switch (action) {
+					case AssemblyAction.CopyUsed:
+						if (IsUsedAssembly (a)) {
```

Comment:
I think this check is unnecessary because an unused `CopyUsed` assembly would already have been marked `Delete`.

Reply:
This is for the case where you have an assembly that is `copyused` and it's used (required in the output) and references another assembly that was removed. We cannot do a simple copy in such cases.

Reply:
Right - I am saying that if we get here for the `CopyUsed` action, we already know `IsUsedAssembly` is true - otherwise `RemoveUnmarkedAssembly` would have set the action to `Delete` earlier. You could get rid of the check and just fall through to the `Copy` case.

Reply:
Fixed